tests/unit/parser/test_float_literals.py

```
@@ -0,0 +1,150 @@
+    def test_leading_dot_with_exponent(self, parser, expr, expected):
+        lit = parse_return_literal(parser, expr)
+        assert isinstance(lit, Literal)
+        assert lit.value == pytest.approx(expected, rel=1e-10)
+
+
+@pytest.mark.unit
+class TestFloatOverflow:
+    """Float literals that overflow IEEE 754 double precision raise an error."""
+
+    def test_positive_overflow_raises(self, parser):
+        with pytest.raises(Exception, match="[Oo]verflow"):
```

Ruff (0.15.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)